fs/span-manager/span_manager.go

```diff
@@ -145,10 +168,11 @@ func (m *SpanManager) resolveSpan(spanID compression.SpanID) error {
 
 // GetContents returns a reader for the requested contents. The contents may be
 // across multiple spans.
-func (m *SpanManager) GetContents(startUncompOffset, endUncompOffset compression.Offset) (io.Reader, error) {
+func (m *SpanManager) GetContents(startUncompOffset, endUncompOffset compression.Offset) (io.ReadCloser, error) {
 	si := m.getSpanInfo(startUncompOffset, endUncompOffset)
 	numSpans := si.spanEnd - si.spanStart + 1
 	spanReaders := make([]io.Reader, numSpans)
+	spanClosers := make([]io.Closer, numSpans)
```

Comment:
nit: can we change the type of `spanReaders` from `[]io.Reader` to `[]io.ReadCloser` so we don't have to track 2 arrays?

Reply:
Yeah, I tried that at first but the compiler complained about mismatched types since `io.MultiReader` strictly accepts `[]io.Reader`.

Reply:
Oh right. Slices don't work like that.